### PR TITLE
Gateway protect + symcheck 

### PR DESCRIPTION
**Pull request by Aqualex**

### CODE REVIEW

**Comment by ThomasSmyth** [comment]
You've ignored everything I mentioned in my comment this morning.

File: appconfig/settings/checker.q [hunk 1]
```diff
@@ -1,3 +1,15 @@
 \d .symcheck
-tm1:0D00:05:00.000000
-wn1:0D00:05:00.000000
+tm1:0D00:10:00.000000
+wn1:0D00:10:00.000000
```
Comment (ThomasSmyth) [inline appconfig/settings/checker.q]: Semi colons.

File: code/checks/symcheck.q [hunk 2]
```diff
@@ -1,47 +1,60 @@
...
 
+notpconnected:{0=count select from .sub.SUBSCRIPTIONS where proctype in .symcheck.tickerplanttypes,active};
+
+getPort:{[x]
+  :.sub.getsubscriptionhandles[x;();()!()] 
+ }
+
 subscribe:{                                                                                     //subscribe to rdb
-  if[count s:.sub.getsubscriptionhandles[`rdb;();()!()];                                        //get handle
+  if[count s:getPort[`rdb];                                                                     //get handle
```
Comment (ThomasSmyth) [inline code/checks/symcheck.q]: Remove squares.

File: code/checks/symcheck.q [hunk 3]
```diff
@@ -1,47 +1,60 @@
...
    subproc:first s;
    .lg.o[`subscribe;"subscribing to ", string subproc`procname];                                //if got handle successfully, subsribe to tables
-   :.sub.subscribe[`quote`trade`trade_iex`quote_iex;`;0b;0b;subproc];
+   :.sub.subscribe[t::((subproc`w)"tables[]")except ignorelist;`;0b;0b;subproc];
   ]
```
Comment (ThomasSmyth) [inline code/checks/symcheck.q]: Better to avoid sending strings over the handle.

File: code/checks/symcheck.q [hunk 4]
```diff
@@ -1,47 +1,60 @@
...
+.email.connect`url`user`password`from`usessl`debug#.email;                                      //create email connection, namespace read in from default.q 
 
-.email.connect`url`user`password`from`usessl`debug#.email;                                      //create email connection, namespace read in from default.q
+tablist:.symcheck.t;
 
-tablist:`quote`trade`trade_iex`quote_iex;
+timeAdjust:{
+  SysTime:first "T"$system"date '+%T'";
+  RDBTime:(first .symcheck.getPort[`rdb]`w)"exec last`time$time from trade";
+  timeAdj:SysTime-RDBTime;:`time$timeAdj
```
Comment (ThomasSmyth) [inline code/checks/symcheck.q]: Semi colon.

File: code/checks/symcheck.q [hunk 5]
```diff
@@ -1,47 +1,60 @@
...
 
-.email.connect`url`user`password`from`usessl`debug#.email;                                      //create email connection, namespace read in from default.q
+tablist:.symcheck.t;
 
-tablist:`quote`trade`trade_iex`quote_iex;
+timeAdjust:{
+  SysTime:first "T"$system"date '+%T'";
+  RDBTime:(first .symcheck.getPort[`rdb]`w)"exec last`time$time from trade";
+  timeAdj:SysTime-RDBTime;:`time$timeAdj   
+ }
```
Comment (ThomasSmyth) [inline code/checks/symcheck.q]: Semi colon.

File: code/checks/symcheck.q [hunk 6]
```diff
@@ -57,15 +70,14 @@ symgrab:{
     grab:{[x]select last time by sym,tab:"s"$x from x};
     data:(first exec w from .sub.getsubscriptionhandles[`rdb;();()!()])(grab;x);                //select data from rdb (using handle number gained from .sub function)
     `symtab upsert data;
-  }each tablist;
+  }each tablist
```
Comment (ThomasSmyth) [inline code/checks/symcheck.q]: Bring back the semi colon.

File: code/checks/symcheck.q (hunk 2 above)
Comment (JuliaPartridge) [inline code/checks/symcheck.q]: Why not use your variable RDBPort on line 9? Is getPort used for other processes? If not, it seems unnecessary to have

**Pull request merged**